cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java

```diff
@@ -43,24 +48,37 @@ public class ApplicationRegistrationStage extends AbstractStage<ApplicationWithP
   private final Store store;
   private final UsageRegistry usageRegistry;
   private final OwnerAdmin ownerAdmin;
+  private final MetricsCollectionService metricsCollectionService;
 
-  public ApplicationRegistrationStage(Store store, UsageRegistry usageRegistry, OwnerAdmin ownerAdmin) {
+  public ApplicationRegistrationStage(Store store, UsageRegistry usageRegistry, OwnerAdmin ownerAdmin,
+                                      MetricsCollectionService metricsCollectionService) {
     super(TypeToken.of(ApplicationWithPrograms.class));
     this.store = store;
     this.usageRegistry = usageRegistry;
     this.ownerAdmin = ownerAdmin;
+    this.metricsCollectionService = metricsCollectionService;
   }
 
   @Override
   public void process(ApplicationWithPrograms input) throws Exception {
     ApplicationSpecification applicationSpecification = input.getSpecification();
+    ApplicationId applicationId = input.getApplicationId();
     Collection<ApplicationId> allAppVersionsAppIds =
-      store.getAllAppVersionsAppIds(input.getApplicationId().getAppReference());
+      store.getAllAppVersionsAppIds(applicationId.getAppReference());
     boolean ownerAdded = addOwnerIfRequired(input, allAppVersionsAppIds);
     ApplicationMeta appMeta = new ApplicationMeta(applicationSpecification.getName(), input.getSpecification(),
                                                   input.getChangeDetail());
     try {
-      store.addApplication(input.getApplicationId(), appMeta);
+      DeployEventType deployEventType = store.addApplication(input.getApplicationId(), appMeta);
+      if (deployEventType.equals(DeployEventType.NEW)) {
+        // increment metric : event.deploy.new.count
+        emitMetrics(applicationId.getNamespace(), applicationId.getApplication(),
+                    Constants.Metrics.AppMetadataStore.EVENT_DEPLOY_NEW_COUNT);
+      } else {
```

Comment:
this means this edit count will include upgrades. Is that the intended behavior, or are they supposed to be mutually exclusive?

Reply:
Oh, i think the intent is to keep them separate. I am thinking I'll set the DeployEvent to UPGRADE at L633 in AppLifecycleService https://github.com/cdapio/cdap/pull/14956/files#diff-bd61e1bd8f120c6108e9b2346f1b32ab44bfc6d476617641ba2e9458166d08b4R633 and then include the check to  filter those events out. Is that good?

Reply:
it's fine to set some flag or something to say it's an upgrade, but I don't think we'll be able to set a DeployEvent in the other scenarios, since we don't know if it's a new deploy or an edit until we call store.addApplication().